### ISSUE TITLE
Check for nulls

### ISSUE THREAD

**Issue by memorex386** [ManueGE/Boxfit#3]
In the generated BoxFit Serializer classes for Kotlin you assign the variables to the object whether it is null or not - I think you need to check for nulls and not apply if null.   Currently you have -

```  
protected void merge(Json json, CustomObject object, BoxStore boxStore) {
 
    if (json.has("CreatedBy")) {
      Long value = json.getLong("CreatedBy");
      com.pakcage.CustomObjectKtBridge.INSTANCE.setCreatedBy(object, value);
    }
```

the problem is that `setCreatedBy` may not accept a null value which in turn causes this to crash.  I think you need to refactor `setCreatedBy` (and all generated methods) from this -

```
    fun setCreatedBy(obj: ServiceRequest, value: Long) {
        obj.createdBy = value
    }
```
to this (notice value is now nullable)

```
    fun setCreatedBy(obj: ServiceRequest, value: Long?) {
     if (value != null)   obj.createdBy = value
    }
```

Kotlin allows for default values so we can safely assume that they have assigned a value for the variable (unless they used `lateinit` but they should understand that.

I really like what you've done and I might branch off what you've done and do this fix quick if that helps.  Thanks!

**Comment by ManueGE** [comment]
If I understand ok, I wouldn't say that it is a **Boxfit** issue. I mean, if the `createdBy` property can't be `null`, the json value should never be `null`. Also, if the json value may be `null`, why simply don't set the field as nullable too?.

IMHO, the client is reponsible of assigning the right types to the models. It would be the same case as if you define a property as `toOne` but the json gets a `JSONArray`, I'd say the problem here would be that the types are not properly defined. 

In any case, I can think in a quick solution. You can add a transformer to your `BoxfitClass` annotation.  You can create a `JSONTransformer` class that remove the `createdBy` key from the json in case it is null.

This wouldn't be a difficult thing to fix, I am just wondering if it the library should take care of this or not  :).

**Comment by memorex386** [comment]
I think I agree - the real issue is here is that our server side data and app side data are not 100% inlined.  The real fix should be that these two data sets aline correctly - but we all now when you have two different groups of people in a big company it can be easier said than done :)

It would be nice to at least have the option to handle these edge cases.  Maybe making a user add a JSONTransformer to every case like this would encourage everyone to actually get the issues fixed on their side :)

**Comment by memorex386** [comment]
@ManueGE I think that would be great.  After reading what you said I agree that by default it should be `false` and you can still have the option of adjusting that

**Comment by ManueGE** [comment]
Fixed in [release 0.0.3](https://github.com/ManueGE/Boxfit/releases/tag/0.0.3).

You can annotate a field with `@FromJsonIgnoreNull` (aside `@BoxfitField`)  to achieve this behavior.

Closing issue.
